mtags/src/main/scala/scala/meta/internal/mtags/CoursierComplete.scala

```diff
@@ -0,0 +1,66 @@
+package scala.meta.internal.mtags
+
+import scala.collection.JavaConverters._
+
+import scala.meta.internal.tokenizers.Chars
+
+object CoursierComplete {
+  def complete(dependency: String): List[String] = {
+    val scalaVersion = BuildInfo.scalaCompilerVersion
+    val api = coursierapi.Complete
+      .create()
+      .withScalaVersion(BuildInfo.scalaCompilerVersion)
+      .withScalaBinaryVersion(
+        if (scalaVersion.startsWith("3")) "3"
+        else scalaVersion.split('.').take(2).mkString(".")
+      )
+    def completions(s: String): List[String] =
+      api.withInput(s).complete().getCompletions().asScala.toList
```

Comment:
Isn't coursier API blocking? If yes, this will silently block current thread until completions are resolved.
I haven't looked at code from the broader perspective but this can be fine if this will affect only scala CLI scripts when only typing "using lib".

Reply:
I'm not sure, but it looks like the performance is ok, maybe we can wait and see if it causes issues and then wrap it with some max timeout. 

> if this will affect only scala CLI scripts when only typing "using lib".

It will trigger only in "using lib", but I haven't yet figured out how to check if it's a scala cli script. Is there even a way to do that?

Also, `CoursierComplete` uses caching, so only the first api call will take longer time.

Reply:
Note that coursier can get into a funky state if there's problems with configured resolves, internet access, etc. For example just yesterday I had internet acccess cut off while scala-cli was resolving artifacts (using coursier), which basically locks up the entire Scala-cli BSP and the only way to fix it is to wipe out all Coursier's cache directories. 

﻿﻿
<img width="762" alt="image" src="https://user-images.githubusercontent.com/1052965/191723213-99d9bf7d-94d8-4e95-8dd4-1f1900e06d18.png">

My general point is that for features that are just convenience, rather than main functionality, Metals should aggressively sandbox them, shift to a separate thread, make them async, and slap a hard timeout.

Any risk of locking up Metals for whatever reason can cause really bad experience.